includes/admin/class-wcmp-export.php

```diff
@@ -1566,45 +1571,135 @@ private function printLabels(array $order_ids, array $shipment_ids, int $offset)
-        $return  = $this->add_shipments($order_ids, $process);
+        return $this->setFeedbackForClient($print, $offset, $orderIds, $return ?? []);
+    }
 
+    /**
+     * @param array $orderIds
+     *
+     * @return array
+     *
+     * @throws \Exception
+     */
+    private function saveOrderCollection(array $orderIds): array
+    {
+        $apiKey          = $this->getSetting(WCMYPA_Settings::SETTING_API_KEY);
+        $orderCollection = (new OrderCollection())->setApiKey($apiKey);
+        $locale          = get_option('WPLANG', 'nl_NL');
+
+        foreach ($orderIds as $orderId) {
+            $wcOrder         = WCX::get_order($orderId);
+            $orderSettings   = new OrderSettings($wcOrder);
+            $deliveryOptions = $orderSettings->getDeliveryOptions();
+
+            $order = (new Order())
+                ->setStatus($wcOrder->get_status())
+                ->setDeliveryOptions($deliveryOptions)
+                ->setInvoiceAddress($orderSettings->getBillingRecipient())
+                ->setRecipient($orderSettings->getShippingRecipient())
+                ->setLanguage($locale)
+                ->setType(null) // what do we want here?
+                ->setOrderDate($wcOrder->get_date_created())
+                ->setExternalIdentifier($orderId);
+                //->setFulfilmentPartnerIdentifier($uuid)
+
+            $orderLines = new Collection();
+
+            foreach ($wcOrder->get_items() as $wcOrderItem) {
+                $orderLine = new OrderLineFromWooCommerce($wcOrderItem);
+
+                $orderLines->push($orderLine);
+            }
+
+            $order->setOrderLines($orderLines);
+            $orderCollection->push($order);
+        }
+
+        return $this->updateOrderMetaByCollection($orderCollection->save());
+    }
+
+    /**
+     * @param \MyParcelNL\Sdk\src\Collection\Fulfilment\OrderCollection $orderCollection
+     *
+     * @return array
+     */
+    private function updateOrderMetaByCollection(OrderCollection $orderCollection): array
+    {
+        $currentDateTime = (new DateTime())->format(AbstractOrder::DATE_FORMAT_FULL);
+
+        foreach ($orderCollection->getIterator() as $order) {
+            $orderId = $order->getExternalIdentifier();
+            $wcOrder = WCX::get_order($orderId);
+            $value   = [
+                WCMYPA_Admin::META_PPS_EXPORTED    => true,
+                WCMYPA_Admin::META_PPS_UUID        => $order->getUuid(),
+                WCMYPA_Admin::META_PPS_EXPORT_DATE => $currentDateTime,
+            ];
+
+            if (! add_post_meta($orderId, WCMYPA_Admin::META_PPS, $value)) {
+                $this->errors[] = sprintf(__('error_pps_export_feedback', 'woocommerce-myparcel'), $orderId);
+            }
+
+            WCMP_API::updateOrderStatus($wcOrder);
+        }
+
+        return [
+            'success' => sprintf(
+                __('myparcel_orders_now_in_backoffice', 'woocommerce-myparcel'),
```

Comment:
dit misschien prefixen met `message` of `notification` oid. Er hoeft wmb ook geen "myparcel" in, ik zou hem zoiets noemen: `message_orders_exported_successfully`